src/storage-controller/src/lib.rs

```diff
@@ -968,18 +968,20 @@ where
                 // We don't care about the dependency since when the write
                 // frontier is empty. In that case, no-one can write down any
                 // more updates.
-                mz_ore::soft_assert_or_log!(
-                    write_frontier.elements() == &[T::minimum()]
-                        || write_frontier.is_empty()
-                        || PartialOrder::less_than(&dependency_since, write_frontier),
-                    "dependency since has advanced past dependent ({id}) upper \n
+                if description.primary.is_none() {
```

Comment:
I decided for this check instead of the perhaps more obvious `if matches!(data_source, Ingestion(..) | IngestionExport { .. })` so we remember to remove this again when/if we get rid of the `primary` stuff.